crates/sui-metrics/src/lib.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use once_cell::sync::OnceCell;
+use prometheus::{register_int_gauge_vec_with_registry, IntGaugeVec, Registry};
+use tap::TapFallible;
+
+use tracing::warn;
+
+pub use scopeguard;
+
+#[derive(Debug)]
+pub struct Metrics {
+    pub tasks: IntGaugeVec,
+    pub futures: IntGaugeVec,
+}
+
+impl Metrics {
+    fn new(registry: &Registry) -> Self {
+        Self {
+            tasks: register_int_gauge_vec_with_registry!(
+                "monitored_tasks",
+                "Number of running tasks per callsite.",
+                &["callsite"],
+                registry,
+            )
+            .unwrap(),
+            futures: register_int_gauge_vec_with_registry!(
+                "monitored_futures",
+                "Number of pending futures per callsite.",
+                &["callsite"],
+                registry,
+            )
+            .unwrap(),
+        }
+    }
+}
+
+static METRICS: OnceCell<Metrics> = OnceCell::new();
+
+pub fn init_metrics(registry: &Registry) {
+    let _ = METRICS
+        .set(Metrics::new(registry))
+        // this happens many times during tests
+        .tap_err(|_| warn!("init_metrics registry overwritten"));
+}
+
+pub fn get_metrics() -> Option<&'static Metrics> {
+    METRICS.get()
+}
+
+#[macro_export]
+macro_rules! monitored_future {
+    ($fut: expr) => {{
+        monitored_future!(futures, $fut)
+    }};
+
+    ($metric: ident, $fut: expr) => {{
+        let name = format!("{}_{}", file!(), line!());
```

Comment:
We can do the following and eliminate the overhead of the format each time we spawn something.

```suggestion
        const LOCATION: &str = concat!(file!(), ':', line!());
```

Reply:
done